homeassistant/components/xiaomi_ble/__init__.py

```diff
@@ -31,9 +32,35 @@ def process_service_info(
     entry: config_entries.ConfigEntry,
     data: XiaomiBluetoothDeviceData,
     service_info: BluetoothServiceInfoBleak,
+    device_registry: DeviceRegistry,
 ) -> SensorUpdate:
     """Process a BluetoothServiceInfoBleak, running side effects and returning sensor data."""
     update = data.update(service_info)
+    if update.events:
+        address = service_info.device.address
+        for device_key, event in update.events.items():
+            sensor_device_info = update.devices[device_key.device_id]
+            device = device_registry.async_get_or_create(
+                config_entry_id=entry.entry_id,
+                identifiers={(DOMAIN, address)},
```

Comment:
```suggestion
                identifiers={(BLUETOOTH_DOMAIN, address)},
```

Since multiple integrations can use the same bluetooth device historically we use the bluetooth domain for identifiers before we added bluetooth connections. Since the RSSI entity will be created with this it needs to match here